static.conversion/public/conversion_tool.json

```
@@ -23,7 +23,7 @@
 
                 "returnType": "xmi",
 
-                "path": "http://127.0.0.1:8069/services/FlexmiToXmi"
+                "path": "{{BASE-URL}}:8069/services/FlexmiToXmi"
```

This doesn't look right (and equally for the changes in the other files): `{{BASE-URL}}` will be replaced with the full URL from which the tool configuration file has been loaded. I would expect this to include the port number (and, on the hosted platform, the path prefix for the respective tool). So I would rather expect something like:

```suggestion
                "path": "{{BASE-URL}}/services/FlexmiToXmi"
```

Have you tested this change in any way?

---

We do need to have the port number because it will get extracted by the script in ToolsManager.js and sent to the discovery service. The discovery service then returns the relevant path in the URL. If port is not present, the correct URL path cannot be retrieved.

---

I didn't realise you're using the discovery service for URLs inside tool specs, too. I assumed you would only use it to translate URLs in activity specifications so that you know where to load the tool specification from. Once you have that, you should then be able to rewrite `{{BASE-URL}}` templates inside tool specifications without reference to the discovery service (though you might need to rewrite references to other tools). 

Would be good to create, and link from here, the companion PR in https://github.com/mdenet/educationplatform so the changes here can be reviewed in context of that code.

---

The reason behind it is that the web user interface makes HTTP requests from end-user's machine to these URLs after loading tool configurations and since the containers are not aware of their exposed ports internally, there is no way to map from URL path to port.